File: src/main/java/com/myudog/myulib/api/events/ComponentAddedEvent.java
package com.myudog.myulib.api.events;

import com.myudog.myulib.api.core.ecs.IComponent;
import com.myudog.myulib.api.core.event.IEvent;

public class ComponentAddedEvent implements IEvent {
    private final int entityId;
    private final IComponent component;

    public ComponentAddedEvent(int entityId, IComponent component) {
        this.entityId = entityId;
        this.component = component;
    }

    public int getEntityId() {
        return entityId;
    }

    public IComponent getComponent() {
        return component;
    }
}

File: src/main/kotlin/com/myudog/myulib/api/ecs/event/ComponentAddedEvent.kt
package com.myudog.myulib.api.ecs.event

import com.myudog.myulib.api.event.Event
import com.myudog.myulib.api.ecs.Component

/**
 * 當實體被成功添加某個 Component 時觸發的事件。
 * 這是一個單純的通知事件（通常不會失敗或被攔截），因此不需要實作 FailableEvent。
 */
data class ComponentAddedEvent(
    val entityId: Int,
    val component: Component
) : Event